plugins/modules/aci_fabric_pod_external_tep.py

```diff
@@ -0,0 +1,321 @@
+  name_alias:
+    description:
+    - The alias for the current object. This relates to the nameAlias field in ACI.
+    type: str
+  pool:
+    description:
+    - The subnet IP address pool for the External TEP.
+    - Must be valid IPv4 or IPv6 and include the subnet mask.
+    - Example 192.168.1.0/24 or 2001:db8:abcd:0012::0/64
+    type: str
+    aliases: [ ip, ip_address, tep_pool ]
+  reserve_address_count:
+    description:
+    - Indicates the number of IP addresses that are reserved from the start of the subnet.
+    type: int
+    aliases: [ address_count ]
+  status:
+    description:
+    - State of the External TEP C(active) or C(inactive).
+    - An External TEP can only be deleted when the state is inactive.
+    - The APIC defaults to C(active) when unset during creation.
+    type: str
+    choices: [ active, inactive ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.aci.aci
+- cisco.aci.annotation
+- cisco.aci.owner
+
+notes:
+- The C(Fabric Setup Pod Policy) must exist before using this module in your playbook.
+  The M(cisco.aci.aci_fabric_setup_pod) can be used for this.
+seealso:
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC class B(fabric:ExtRoutablePodSubnet).
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- Samita Bhattacharjee (@samitab)
+"""
+
+EXAMPLES = r"""
+- name: Add an External TEP to a pod fabic setup policy
+  cisco.aci.aci_fabric_pod_external_tep:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    pod_id: 2
+    pool: 10.6.1.0/24
+    address_count: 5
+    status: active
+    state: present
+  delegate_to: localhost
+
+- name: Change an External TEP state on a pod fabic setup policy to inactive
+  cisco.aci.aci_fabric_pod_external_tep:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    pod_id: 2
+    pool: 10.6.1.0/24
+    status: inactive
+    state: present
+  delegate_to: localhost
+
+- name: Delete an External TEP on a pod fabic setup policy
+  cisco.aci.aci_fabric_pod_external_tep:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    pod_id: 2
+    pool: 10.6.1.0/24
+    state: absent
+  delegate_to: localhost
+
+- name: Query the External TEP on a pod fabic setup policy
+  cisco.aci.aci_fabric_pod_external_tep:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    pod_id: 2
+    pool: 10.6.1.0/24
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query External TEPs on all pod fabic setup policies
+  cisco.aci.aci_fabric_pod_external_tep:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
```

Comment:
suggestion from @sajagana was to put absent to the end

```suggestion
- name: Query the External TEP on a pod fabic setup policy
  cisco.aci.aci_fabric_pod_external_tep:
    host: apic
    username: admin
    password: SomeSecretPassword
    pod_id: 2
    pool: 10.6.1.0/24
    state: query
  delegate_to: localhost
  register: query_result

- name: Query External TEPs on all pod fabic setup policies
  cisco.aci.aci_fabric_pod_external_tep:
    host: apic
    username: admin
    password: SomeSecretPassword
    state: query
  delegate_to: localhost
  register: query_result

- name: Delete an External TEP on a pod fabic setup policy
  cisco.aci.aci_fabric_pod_external_tep:
    host: apic
    username: admin
    password: SomeSecretPassword
    pod_id: 2
    pool: 10.6.1.0/24
    state: absent
  delegate_to: localhost
```